### DISCUSSION TITLE
post/5/

### DISCUSSION THREAD

**Discussion by giscus** [HoshinoRei/blog#14] [General]
# post/5/

求生之路（Left 4 Dead 2）是小玲平常玩的游戏之一，这款游戏虽然发售了 13 年，但至今依然有不小的热度。但是小玲在游玩时发现，哪怕是在香港的官服，

https://blog.bling.moe/post/5/

**Comment by emoker** [comment]
大佬文档写的很清楚，我刚接触 L4D2 开服想请教下：
1. 如果像您这样用docker-compose 开服对CPU要求高吗，我自己有个1核的服务器，看别人说一个纯净服是够的，不知道用docer-compose 会不会出现cpu 不够的情况。
2. 如果L4D2游戏有更新容器能不能自动处理？还是要要自己更新文件？
3. 这样的服务器要装插件的话是不是往addons 添加插件文件就好了？

**Reply by yiksun** [reply]
1. 如果像您这样用docker-compose 开服对CPU要求高吗，我自己有个1核的服务器，看别人说一个纯净服是够的，不知道用docer-compose 会不会出现cpu 不够的情况。
*docker-compose只是代替docker run命令, 并不会对运行的容器产生额外消耗, 只是安装docker-compose可能需要额外的硬盘空间*
*求生之路4人服对系统需求很低, 我用腾讯云最基础的轻量服务器(2核/2G/4M)无压力运行*
2. 如果L4D2游戏有更新容器能不能自动处理？还是要要自己更新文件？
*如果官方游戏更新了, 我们有三种更新自己服务器的方法*
*1. 重新拉博主的镜像, 前提是博主已经更新了镜像到最新的版本*
*2. 如果博主没有及时更新, 可以在自己的服务器用博主的Dockerfile自行构建一个最新的服务器镜像*
*3. 在自己的服务器上更新, 大概思路是attach近运行中的求生之路服务器, 用steamcmd更新服务器文件, 具体命令参考博主的Dockerfile* (参考 https://www.bilibili.com/read/cv13455610/)
3. 这样的服务器要装插件的话是不是往addons 添加插件文件就好了？
*参考 https://www.bilibili.com/read/cv16824745/*

**Reply by emoker** [reply]
非常感谢 😃

**Reply by HoshinoRei** [reply]
1. 可以吧，我曾经用过一个腾讯云的轻量服务器开，可以 4 个人 100 tick。
2. 可以用 [watchtower](https://containrrr.dev/watchtower/) 监控镜像的更新。我有空的话会及时更新游戏版本的，如果没更新可以提醒一下我。
3. 可以往 addons 里添加插件，但我自己的话会再写一个 Dockerfile 继承目前这个镜像。

**Comment by iicemeta** [comment]
大佬服务器默认是100tick吗？不是的话应该用什么模组？

**Reply by HoshinoRei** [reply]
默认 30 tick，需要 sourcemod、tickrate enabler、l4dtoolz。

**Reply by iicemeta** [reply]
老哥你的项目的docker hub那个l4d2版本没更新，我用你的项目重新构建了

[l4d2-docker](https://github.com/24works/l4d2-docker)

**Reply by HoshinoRei** [reply]
了解。

**Comment by Xenoxenox** [comment]
安装MetaMOD后，需要将./addons/metamod.vdf修改为：
```
"Plugin"
{
	"file"	"./addons/metamod/bin/server"
}
```
相对于在addons前面加上”./“，与docker-compose.yml 定义的volumes映射表相匹配，否则l4d2服务器会找不到addons目录下的MetaMOD插件

**Reply by Xenoxenox** [reply]
相对于-->相当于
